scripts/vnet_route_check.py

```diff
@@ -0,0 +1,339 @@
+    ''' Returns dictionary of VNET routes configured per each VNET in ASIC_DB.
+    Format: { <vnet_name>: { 'routes': [ <pfx/pfx_len> ], 'vrf_oid': <oid> } }
+    '''
+    db = ConfigDBConnector()
+    db.db_connect('ASIC_DB')
+
+    vnet_vrfs = get_vrf_entries()
+    vnet_vrfs_oids = [vnet_vrfs[k] for k in vnet_vrfs]
+    
+    vnet_intfs = get_vnet_intfs()
+
+    vrf_oid_to_vnet_map = {}
+
+    for vnet_name, vnet_rifs in vnet_intfs.items():
+        for vnet_rif, vrf_oid in vnet_vrfs.items():
+            if vnet_rif in vnet_rifs:
+                vrf_oid_to_vnet_map[vrf_oid] = vnet_name
+
+    routes_db_keys = db.get_keys('ASIC_STATE:SAI_OBJECT_TYPE_ROUTE_ENTRY', False)   
+    vnet_routes = {}
+
+    for route_db_key in routes_db_keys:
+        route_attrs = route_db_key.lower().split('\"', -1)
+        # route_attrs[11] - VRF OID for the VNET route
+        # route_attrs[3] - VNET route IP subnet
+        vrf_oid = route_attrs[11]
+        ip_addr = route_attrs[3]
+
+        if vrf_oid in vnet_vrfs_oids:
+            if vrf_oid_to_vnet_map[vrf_oid] not in vnet_routes:
+                vnet_name = vrf_oid_to_vnet_map[vrf_oid]
+
+                vnet_routes[vnet_name] = {}
+                vnet_routes[vnet_name]['routes'] = []
+                vnet_routes[vnet_name]['vrf_oid'] = vrf_oid
+
+            vnet_routes[vnet_name]['routes'].append(ip_addr)
+
+    filter_out_vnet_ip2me_routes(vnet_routes)
+
+    return vnet_routes
+
+
+def get_vnet_routes_diff(routes_1, routes_2):
+    ''' Returns all routes present in routes_2 dictionary but missed in routes_1
+    Format: { <vnet_name>: { 'routes': [ <pfx/pfx_len> ] } }
+    '''
+
+    routes = {}
+
+    for vnet_name, vnet_attrs in routes_2.items():
+        if vnet_name not in routes_1:
+            routes[vnet_name] = routes
+        else:
+            for vnet_route in vnet_attrs['routes']:
+                if vnet_route not in routes_1[vnet_name]['routes']:
+                    if vnet_name not in routes:
+                        routes[vnet_name] = {}
+                        routes[vnet_name]['routes'] = []
+                    routes[vnet_name]['routes'].append(vnet_route)
+
+    return routes
+
+
+def get_sdk_vnet_routes_diff(routes):
+    ''' Returns all routes present in routes dictionary but missed in SAI/SDK
+    Format: { <vnet_name>: { 'routes': [ <pfx/pfx_len> ], 'vrf_oid': <oid> } }
+    '''
+    routes_diff = {}
+
+    res = os.system('docker exec syncd test -f /usr/bin/vnet_route_check.py')
+    if res != 0:
+        return routes_diff
+
+    for vnet_name, vnet_routes in routes.items():
+        vnet_routes = routes[vnet_name]["routes"]
+        vnet_vrf_oid = routes[vnet_name]["vrf_oid"]
+
+        res = os.system('docker exec syncd "/usr/bin/vnet_route_check.py {} {}"'.format(vnet_vrf_oid, vnet_routes))
+        if res:
+            routes_diff[vnet_name] = {}
+            routes_diff[vnet_name]['routes'] = res
+
+    return routes_diff
+
+
+def main():
+
```

Comment:
Since the script is integrated to monit, we don't want to generate any false alarms for other deployments/platforms. Suggest, the script should first check if there are any VNET config and if not, just return OK and not proceed.

Reply:
@volodymyrsamotiy , can you take a look at this comment?